src/Controller/CRUDController.php

```diff
@@ -66,11 +68,29 @@ class CRUDController extends Controller
      */
     private $templateRegistry;
 
-    public function setContainer(?ContainerInterface $container = null): void
+    public function setContainer(ContainerInterface $container): ?ContainerInterface
     {
-        $this->container = $container;
+        $result = parent::setContainer($container);
 
         $this->configure();
+
+        return $result;
+    }
+
+    public static function getSubscribedServices()
+    {
+        return array_merge(parent::getSubscribedServices(), [
```

Comment:
Since the keys are strings and the values are scalars, I think we could use the `+` operator instead of `array_merge()`.
BTW, do you think we could use DI in the future for these services? I was not following this PR, so I don't know why we need to use this method.

Reply:
This was discussed on my previous attempt (using DI). You can read about it here: https://github.com/sonata-project/SonataAdminBundle/pull/6472#discussion_r502885341

About the use of array_merge, I copied the default example of Symfony: https://symfony.com/doc/current/service_container/service_subscribers_locators.html#including-services

Should I change it ?

Reply:
>This was discussed on my previous attempt (using DI). You can read about it here: #6472 (comment)

Thank you, I'll check there.

>About the use of array_merge, I copied the default example of Symfony: >https://symfony.com/doc/current/service_container/service_subscribers_locators.html#including-services
>
>Should I change it ?

IMO, if we don't require the features of `array_merge()` (I think we don't), we can save a function call by using `+` instead.

Reply:
Fixed

Reply:
Thanks! In order to keep the precedence like it was with `array_merge()`, I think we should flip the operands: https://3v4l.org/h66Tg.